core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/hybrid/HybridRoleManager.java

```diff
@@ -433,6 +434,87 @@ public void updateUserListOfHybridRole(String roleName, String[] deletedUsers, S
         }
     }
 
+    /**
+     * Update group list of role.
+     *
+     * @param roleName      Role name.
+     * @param deletedGroups Deleted groups.
+     * @param newGroups     New groups.
+     * @throws UserStoreException UserStoreException.
+     */
+    public void updateGroupListOfHybridRole(String roleName, String[] deletedGroups, String[] newGroups)
+            throws UserStoreException {
+
+        String sqlStmt1 = HybridJDBCConstants.REMOVE_GROUP_FROM_ROLE_SQL;
+        String sqlStmt2 = HybridJDBCConstants.ADD_GROUP_TO_ROLE_SQL;
+
+        try (Connection dbConnection = DatabaseUtil.getDBConnection(dataSource)) {
+
+            String primaryDomainName = getMyDomainName();
+            if (primaryDomainName != null) {
```

Comment:
why is this primaryDomainName? Can be differ

Reply:
Changed the variable name.